pyxplr/explore_missing.py

```diff
@@ -3,44 +3,80 @@
 @author: Braden Tam
 Implementation of the explore_missing function in the pyxplr package.
 """
+import numpy as np
+import pandas as pd
 
-def explore_missing(dataframe, num_rows = 1):
+def explore_missing(data, num_rows = 0, type = "location"):
     """
-    explore_missing will return 2 tables: 1 table of exactly where there is 
-    missing data and another table how much data is missing for each feature.
+    explore_missing will explore missing observations within data. It will
+    return 1 of 2 tables: 1 table of exactly where there is missing data or 
+    another table showing how many observations are missing, and the 
+    proportion of how much data is missing for each feature. 
 
     Arguments
     ---------
-    dataframe : pandas.DataFrame
+    data : pandas.DataFrame
         The target dataframe to explore
     num_rows : integer
         The number of rows above and below the missing value to output 
+    table: str
+        The desired type of output (location or count)
 
     Returns
     -------
     type : pandas.DataFrame
-        The resultant dataframes
+        The resultant dataframe
 
     Raises
     ------
-    None Missing
+    ValueError
         There are no missing values in the dataframe
-
-    Notes
-    -----
-    (Additional notes to be filled during development)
+    TypeError
+        Data must be a pandas DataFrame
+    NameError
+        Type must be either "count" or "location"
 
     Examples
     --------
     >>> test = pd.DataFrame({'col1': [1, 2, None, 3, 4], 'col2': [2, 3, 4, 5, 6]})
-    >>> explore_missing(test)
+    >>> explore_missing(test, num_rows = 1)
         col1  col2
     0     2     3
     1    NaN    4
     2     3     5
-
-        col1  col2
-    0     1     0
+    >>> explore_missing(test, type = "count")
+            Number of missing values	Proportion of missing data
+    col 0                          1                           0.2
+    col 1                          0                             0
 
     """
-    raise NotImplemented
\ No newline at end of file
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError("Data must be a pandas DataFrame")
+
+    if not (type == "count") | (type == "location"):
+        raise NameError('Type must be either "count" or "location"')
+
+    indices = np.where(data.isnull())[0]
+
+    if len(indices) == 0:
+        raise ValueError("There are no missing values in the dataframe")
```

Comment:
```suggestion
        return pd.DataFrame()
```
Not sure if we need to raise error if there are no missing values - maybe just return empty dataframe?

Reply:
If the function just returns an empty dataframe the user may be confused with the output. I think raising an error will be more meaningful to show that there are in fact no missing values in the data and our function would serve no purpose.